libraries/SPISlave/src/SPISlave.cpp

```diff
@@ -93,18 +93,22 @@ void SPISlaveClass::setStatus(uint32_t status)
 }
 void SPISlaveClass::onData(SpiSlaveDataHandler cb)
 {
+    check_function_is_in_iram(cb);
```

Comment:
This doesn't look right, the handler arg is received by value. Same for the other cases below.